src/extension.ts

```diff
@@ -207,17 +207,41 @@ async function autoUpdate() {
     }
 }
 
-function diagnosticCounter() {
+function progressCounter() {
+    const runningProgress: { [index: string]: boolean } = {};
+    const asPercent = (fraction: number): string => `${Math.round(fraction * 100)}%`;
     let runningDiagnostics = 0;
     lc.onReady().then(() => {
-        lc.onNotification(new NotificationType('rustDocument/beginBuild'), function(_f: any) {
+
+        stopSpinner('RLS');
+
+        lc.onNotification(new NotificationType('window/progress'), function (progress: any) {
+            if (progress.done) {
+                delete runningProgress[progress.id];
+            } else {
+                runningProgress[progress.id] = true;
+            }
+            if (Object.keys(runningProgress).length) {
+                const status =
+                    typeof progress.percentage === 'number' ? asPercent(progress.percentage) :
```

Comment:
Can you use an `if` statment rather than these nested ternary operators, it's pretty hard to read as is.

Reply:
Ok.